code/modules/mob/living/silicon/robot/drone/drone.dm

```diff
@@ -154,15 +162,17 @@
 			return
 
 		else
-			user.visible_message("<span class='warning'>\the [user] swipes [user.p_their()] ID card through [src], attempting to shut it down.</span>", "<span class='warning'>You swipe your ID card through \the [src], attempting to shut it down.</span>")
+			var/confirm = alert("Using your ID on a Maintenance Drone will shut it down, are you sure you want to do this?", "Disable Drone", "Yes", "No")
```

Comment:
Doesn't this let you swipe your ID on a drone, keep the popup on-screen, and then click 'yes' later after the drone has moved offscreen?

Also, if a check is added, doesn't this mean the drone can avoid being de-activated by moving away from you really fast?

Reply:
It does, yes.
I thought it would be better to avoid the second situation since otherwise you could never actually deactivate one, unless you got lucky.

It also won't matter for emagged drones, since they can't be deactivated anyway.